activation/activation.go

```diff
@@ -403,26 +328,53 @@ func (b *Builder) buildNIPostChallenge(ctx context.Context) (*types.NIPostChalle
 		return nil, ctx.Err()
 	case <-b.syncer.RegisterForATXSynced():
 	}
+	current := b.currentEpoch()
+	prev, err := b.cdb.GetLastAtx(b.nodeID)
+	if err != nil {
+		if !errors.Is(err, sql.ErrNotFound) {
+			return nil, err
+		}
+	} else if prev.PublishEpoch == current+1 {
+		current += 1
+	}
```

Comment:
```suggestion
	if err != nil && !errors.Is(err, sql.ErrNotFound) {
		return nil, err
	}
	if prev.PublishEpoch == current+1 {
		current += 1
	}
```

Why is the check for `prev.PublishEpoch` necessary? Will the node eagerly create ATXs for more than one epoch in advance?

Reply:
lets say node successfully submitted atx, and calls this function again (might be after restart). we check that it already submitted atx, and therefore needs to wait for +1 round

Reply:
btw the suggestion doesn't work, as we can't call prev is error is not nill

Reply:
Ah yes, you're right regarding the suggested code. 🙂 

But I'm still failing to understand the ´prev.PublishEpoch == current+1` check?

If the prev. ATX is form the current epoch then all is fine and we create one for the next, but if prev. ATX is for the next epoch we create another one for the epoch after that?

Reply:
🤦 never mind we wait until current...